polaris-patterns/.storybook/preview.js

```diff
@@ -1,26 +1,54 @@
-import React from 'react';
+import React, {useState, useEffect} from 'react';
 import {breakpoints} from '@shopify/polaris-tokens';
-import {I18nContext, I18nManager} from '@shopify/react-i18n';
 import {AppProvider} from '@shopify/polaris';
+import {I18nManager, I18nContext} from '@shopify/react-i18n';
 
+import {
+  PolarisPatternsProvider,
+  useI18nManager,
+} from '../src/components/PolarisPatternsProvider';
 import polarisEnTranslations from '../../polaris-react/locales/en.json';
+import patternsEnTranslations from '../locales/en.json';
 
-const i18nManager = new I18nManager({
-  locale: 'en',
-  currency: 'usd',
-  country: 'CA',
-});
+const defaultTranslations = {
+  polaris: polarisEnTranslations,
+  patterns: patternsEnTranslations,
+};
 
 function AppProviderDecorator(Story, context) {
-  const {locale} = context.globals;
-  i18nManager.update({locale, currency: 'usd', country: 'CA'});
+  const {locale, country, currency} = context.globals;
+  const [translations, setTranslations] = useState(defaultTranslations);
+
+  useEffect(() => {
+    const importTranslations = async () => {
+      try {
+        const patternsTranslations = await import(
+          /* webpackChunkName: "Polaris-Patterns-i18n", webpackMode: "lazy-once" */ `../locales/${locale}.json`
+        );
+        const polarisTranslations = await import(
+          /* webpackChunkName: "Polaris-i18n", webpackMode: "lazy-once" */ `../../polaris-react/locales/${locale}.json`
+        );
+        setTranslations({
+          patterns: patternsTranslations.default,
+          polaris: polarisTranslations.default,
+        });
+      } catch {
+        setTranslations(defaultTranslations);
+      }
+    };
+
+    importTranslations();
+  }, [locale]);
```

Comment:
anyone have context on why we recommend using `@shopify/react-i18n` to dynamically load locales instead of just importing them directly like I did above?

<img width="897" alt="Screenshot 2023-04-24 at 6 29 14 PM" src="https://user-images.githubusercontent.com/20652326/234130228-3e70ab31-79d9-4e64-b384-e54af751e480.png">